From 8fbda3237274767e23c9b30b995768d8e42f498e Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:40:09 -0400
Subject: [ethernet] i40e: fix format mismatch in drivers/net/ethernet/intel/i40e/i40e_debugfs.c

Message-id: <1409658042-22689-249-git-send-email-sassmann@redhat.com>
Patchwork-id: 90426
O-Subject: [RHEL7.1 PATCH 248/281] i40e: fix format mismatch in drivers/net/ethernet/intel/i40e/i40e_debugfs.c
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Toralf Foerster <toralf.foerster@gmx.de>

spotted by cppcheck

Signed-off-by: Toralf Foerster <toralf.foerster@gmx.de>
Tested-by: Jim Young <jamesx.m.young@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit efe1ac25d084fd56c5b809634a0444606c2a5cd3)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_debugfs.c b/drivers/net/ethernet/intel/i40e/i40e_debugfs.c
index ec07332..9eaed04 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_debugfs.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_debugfs.c
@@ -1238,7 +1238,7 @@ static ssize_t i40e_dbg_command_write(struct file *filp,
  } else if (strncmp(cmd_buf, "add pvid", 8) == 0) {
   i40e_status ret;
   u16 vid;
-  int v;
+  unsigned int v;
 
   cnt = sscanf(&cmd_buf[8], "%i %u", &vsi_seid, &v);
   if (cnt != 2) {
@@ -1254,7 +1254,7 @@ static ssize_t i40e_dbg_command_write(struct file *filp,
    goto command_write_done;
   }
 
-  vid = (unsigned)v;
+  vid = v;
   ret = i40e_vsi_add_pvid(vsi, vid);
   if (!ret)
    dev_info(&pf->pdev->dev,
-- 
1.7.1